google/cloud/forseti/common/gcp_api/_base_repository.py

```diff
@@ -488,6 +488,25 @@ def execute_query(self, verb, verb_arguments):
     def _execute(self, request):
         """Run execute with retries and rate limiting.
 
+        Args:
+            request (object): The HttpRequest object to execute.
+
+        Returns:
+            dict: The response from the API.
+        """
+        try:
+            return self._execute_request(request)
+        except exceptions.RefreshError as e:
+            # If there is problem refreshing the token, we will recreate a new
+            # Credential object and use that to refresh the request.
+            LOGGER.exception(e)
+            self._credentials, _ = api_helpers.get_google_default_credentials()
+            self._credentials.refresh(request)
```

Comment:
The usage of `refresh()` is a bit inconsistent.  I see that in the `__init__()` of `BaseRepositoryClient` class that we don't use the `refresh()` is not being used.  Why is that?

And if we really need, how about centralizing it in the `get_google_default_credentials()` as well?

Reply:
refresh will take in a request and we don't have the request object in the __init__ method inside the BaseRepositoryClient class, the request is built with method _build_request() in the GCPRepository class